### ISSUE TITLE
wezterm fails to run on unbuntu through nix

### ISSUE THREAD

**Issue by polarmutex** [wez/wezterm#2021]
### What Operating System(s) are you seeing this problem on?

Linux X11

### WezTerm version

wezterm 20220523

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

when I run wezterm from nix on a Ubuntu 20.04 host the window does not come up and an error is displayed

### To Reproduce

run the command wezterm

### Configuration

no config

### Expected Behavior

wezterm opens without error

### Logs

09:55:31.952  INFO   wezterm_mux_server_impl::local > setting up /run/user/1000/wezterm/gui-sock-135220
09:55:32.081  ERROR  wezterm_gui::frontend          > Failed to create window: with_egl_lib failed: with_egl_lib(libEGL.so.1) failed: egl GetDisplay: Failed but with error code: SUCCESS, libEGL.so: libEGL.so: cannot open shared object file: No such file or directory, with_egl_lib(libEGL.so.1) failed: egl GetDisplay: Failed but with error code: SUCCESS, libEGL.so: libEGL.so: cannot open shared object file: No such file or directory

~
❯ ldd wezterm
ldd: ./wezterm: No such file or directory

~
❯ whereis wezterm
wezterm: /nix/store/nhi1pvjyxfzyfd4zaddccv893bm5slr6-user-environment/bin/wezterm

~
❯ ldd /nix/store/nhi1pvjyxfzyfd4zaddccv893bm5slr6-user-environment/bin/wezterm
        linux-vdso.so.1 (0x00007ffe74764000)
        libssl.so.1.1 => /nix/store/v71f11hrf2b8ga7gnl31h46sd7nky73l-openssl-1.1.1o/lib/libssl.so.1.1 (0x00007f714f5c3000)
        libcrypto.so.1.1 => /nix/store/v71f11hrf2b8ga7gnl31h46sd7nky73l-openssl-1.1.1o/lib/libcrypto.so.1.1 (0x00007f714f2d0000)
        libz.so.1 => /nix/store/r90cncsaa519pwqpijg7ii4rkcmwjn6h-zlib-1.2.12/lib/libz.so.1 (0x00007f714f2b2000)
        libc.so.6 => /nix/store/bvy2z17rzlvkx2sj7fy99ajm853yv898-glibc-2.34-210/lib/libc.so.6 (0x00007f714f0b4000)
        /nix/store/bvy2z17rzlvkx2sj7fy99ajm853yv898-glibc-2.34-210/lib/ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2 (0x00007f715063c000)
        libgcc_s.so.1 => /nix/store/bvy2z17rzlvkx2sj7fy99ajm853yv898-glibc-2.34-210/lib/libgcc_s.so.1 (0x00007f714f09a000)
        libm.so.6 => /nix/store/bvy2z17rzlvkx2sj7fy99ajm853yv898-glibc-2.34-210/lib/libm.so.6 (0x00007f714efbf000)
        libdl.so.2 => /nix/store/bvy2z17rzlvkx2sj7fy99ajm853yv898-glibc-2.34-210/lib/libdl.so.2 (0x00007f714efba000)
        libpthread.so.0 => /nix/store/bvy2z17rzlvkx2sj7fy99ajm853yv898-glibc-2.34-210/lib/libpthread.so.0 (0x00007f714efb5000)

### Anything else?

nope

**Comment by MuhammedZakir** [comment]
You should report this in `nixpkgs`'s repo. I don't know Nix, but it looks like there is no `libGL` in build inputs. If you are fine with building yourself, you may want to look at Kitty's `default.nix` file. Latter seems to solve the `UserNotifications` linking error too.

pinging @SuperSandro2000.

[1] https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/terminal-emulators/wezterm/default.nix
[2] https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/terminal-emulators/kitty/default.nix

**Comment by polarmutex** [comment]
i was using my own overlay to build (that included libGL) and still get the same error

**Comment by SuperSandro2000** [comment]
> You should report this in `nixpkgs`'s repo.

Yes, this should always be done unless it is determined that the issue is not related to nix.

> P.S. Browsing through the repo, I found that `libGL` is used in Linux's build inputs and `OpenGL` in macOS'. The same can be seen in Kitty's. I guess this need to be done for WezTerm too?

Please don't copy that hack. Nix' Apple SDK is just to old, so we are out of luck.

> ```
10:08:32.765 ERROR wezterm_gui::frontend > Failed to create window: with_egl_lib failed: with_egl_lib(libEGL.so.1) failed: egl GetDisplay: Failed but with error code: SUCCESS, with_egl_lib(libEGL.so) failed: egl GetDisplay: Failed but with error code: SUCCESS, with_egl_lib(libEGL.so.1) failed: egl GetDisplay: Failed but with error code: SUCCESS, with_egl_lib(libEGL.so) failed:
egl GetDisplay: Failed but with error code: SUCCESS
``` 

That is caused on the handling of graphics drivers which eventually should be revamped. Currently there is an open RFC about that but it could take a while until that is fixed. Until then you either switch fully to NixOS or use nixGL.

---

I think this issue can be closed. The graphics drivers issue is a known issue for every graphics application and as soon as the SDK is updated someone can send a PR against nixpkgs to fix the darwin build.
